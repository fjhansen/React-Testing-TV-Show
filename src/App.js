import React, { useState, useEffect } from "react";
import axios from "axios";
import Dropdown from "react-dropdown";
import parse from "html-react-parser";

import { useFetchShow } from "./api/fetchShow"

import { formatSeasons } from "./utils/formatSeasons";

import Episodes from "./components/Episodes";
import "./styles.css";

export default function App(props) {
  // const [show, setShow] = useState(null);
  // const [seasons, setSeasons] = useState([]);

  const [selectedSeason, setSelectedSeason] = useState("");
  const episodes = seasons[selectedSeason] || [];

  const [seasons, setSeasons] = useFetchShow(episodes)

  // useEffect was here

  const handleSelect = e => {
    setSelectedSeason(e.value);
  };

  if (!props.axiosshow) {
    return <h2>Fetching data...</h2>;
  }

  return (
    <div className="App">
      <img className="poster-img" src={props.show.image.original} alt={props.show.name} />
      <h1>{props.show.name}</h1>
      {parse(props.show.summary)}
      <Dropdown
        options={Object.keys(seasons)}
        onChange={handleSelect}
        value={selectedSeason || "Select a season"}
        placeholder="Select an option"
      />
      <Episodes episodes={episodes} />
    </div>
  );
}
