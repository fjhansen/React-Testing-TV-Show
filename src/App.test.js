import axios from 'axios'
import React from 'react'


import * as rtl from "@testing-library/react";

import "@testing-library/jest-dom/extend-expect";
import App from "./App";

// jest.mock("axios", () => {
//   return {
//     get: jest.fn(() => Promise.resolve({
//       data: {
//         message: ["test"]
//       }
//     }))
//   }
// })

afterEach(rtl.cleanup);

test("<App/> Snapshot of Fetching data", async () => {
const wrapper = rtl.render(<App/>)
await wrapper.findAllByText(/fetching/i)

expect(wrapper.asFragment()).toMatchSnapshot()
})

test("<App/> Snapshot Full", async () => {
  const wrapper = rtl.render(<App/>)
  await wrapper.findAllByText(/stranger/i)
  
  expect(wrapper.asFragment()).toMatchSnapshot()
  })