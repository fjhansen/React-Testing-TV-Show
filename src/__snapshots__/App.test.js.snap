// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> Snapshot Full 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <img
      alt="Stranger Things"
      class="poster-img"
      src="http://static.tvmaze.com/uploads/images/original_untouched/200/501942.jpg"
    />
    <h1>
      Stranger Things
    </h1>
    <p>
      A love letter to the '80s classics that captivated a generation, 
      <b>
        Stranger Things
      </b>
       is set in 1983 Indiana, where a young boy vanishes into thin air. As friends, family and local police search for answers, they are drawn into an extraordinary mystery involving top-secret government experiments, terrifying supernatural forces and one very strange little girl.
    </p>
    <div
      class="Dropdown-root"
    >
      <div
        aria-haspopup="listbox"
        class="Dropdown-control"
      >
        <div
          class="Dropdown-placeholder is-selected"
        >
          Select a season
        </div>
        <div
          class="Dropdown-arrow-wrapper"
        >
          <span
            class="Dropdown-arrow"
          />
        </div>
      </div>
    </div>
    <div
      class="episodes"
      data-testid="episodes"
    />
  </div>
</DocumentFragment>
`;

exports[`<App/> Snapshot of Fetching data 1`] = `
<DocumentFragment>
  <h2>
    Fetching data...
  </h2>
</DocumentFragment>
`;
